Library/ViewModels/ChangeEmailViewModel.swift

```diff
@@ -46,11 +46,17 @@ public protocol ChangeEmailViewModelType {
 public final class ChangeEmailViewModel: ChangeEmailViewModelType, ChangeEmailViewModelInputs,
 ChangeEmailViewModelOutputs {
   public init() {
+    self.dismissKeyboard = Signal.merge(
+      self.textFieldShouldReturnProperty.signal.skipNil()
+        .filter { $0 == .done }
+        .ignoreValues(),
+      self.saveButtonTappedProperty.signal.ignoreValues())
+
     let changeEmailEvent = Signal.combineLatest(
       self.newEmailProperty.signal.skipNil(),
       self.passwordProperty.signal.skipNil()
       )
-      .takeWhen(self.saveButtonTappedProperty.signal)
+      .takeWhen(self.dismissKeyboard)
```

Comment:
Hmm, I think the desired behavior is that the `changeEmailEvent` should fire if *either* the save button is tapped, *or* the current password text field returns, but *only* if the save button is enabled, otherwise it's sending a request to the server without going through validation first.